src/core_plugins/kibana/public/dashboard/dashboard_state_manager.js

```diff
@@ -138,20 +138,23 @@ export class DashboardStateManager {
   }
 
   _handleStoreChanges() {
-    if (this._areStoreAndAppStatePanelsEqual()) {
-      return;
+    let dirty = false;
+    if (!this._areStoreAndAppStatePanelsEqual()) {
+      const state = store.getState();
```

Comment:
Minor nit, but I think it's slightly easier to read if we assign `panels` here instead, since that's what this logic concerned with:

```js
const panels = getPanels(store.getState());
```

I had to read it a few times to make sure we weren't using `state` anywhere, but this way that becomes self-evident.